corehq/apps/callcenter/tasks.py

```diff
@@ -45,29 +44,22 @@ def calculate_indicators():
         indicator_set.get_data()
 
 
-def sync_web_user_usercases_if_applicable(user, domain):
-    if not USH_USERCASES_FOR_WEB_USERS.enabled(domain):
-        return
-    domain_obj = Domain.get_by_name(domain)
-    if domain_obj.usercase_enabled:
-        sync_usercases_task.delay(user._id, domain)
-
-
 def bulk_sync_usercases_if_applicable(domain, user_ids):
     domain_obj = Domain.get_by_name(domain)
     if domain_obj.call_center_config.enabled or domain_obj.usercase_enabled:
         for user_id in user_ids:
             sync_usercases_task.delay(user_id, domain_obj.name)
 
 
-def sync_usercases_if_applicable(user, spawn_task):
-    if settings.UNIT_TESTING and not user.project:
+def sync_usercases_if_applicable(domain, user, spawn_task):
+    domain_obj = Domain.get_by_name(domain) if domain else None
+    if settings.UNIT_TESTING and not domain_obj:
         return
-    if (user.project.call_center_config.enabled or user.project.usercase_enabled):
+    if (domain_obj.call_center_config.enabled or domain_obj.usercase_enabled):
```

Comment:
if `domain_obj` does end up being None, wouldn't this error? I think we would just return early and not do anything in that case?

Reply:
That's right, this would error.  Same goes for the old code though, so I _think_ we should be okay (`user.project` just calls `Domain.get_by_name`, as did the separate web user version).  If it fails, that means there's a real data issue - a user is a member of a domain that doesn't exist, so I'm kinda tempted to keep it in as a hard failure.

Though I do wonder if there's some way this could cause problems when we roll it out more broadly, like a leftover domain membership from a deleted domain or something.  What do you think?

Reply:
the migration actually doesn't go through this code path. It iterates through the domains and gets the users from each domains. As opposed to iterating through all users and getting the domain each user is a member of.

So I don't think it would run into any issues if the user doc has bad data